cli/providers/deploy.go

```diff
@@ -22,6 +23,11 @@ func Deploy(s *service.Service, e *environment.Environment, log *logger.Logger)
 	if err := e.Validate(); err != nil {
 		return err
 	}
-
-	return aws.Deploy(s, e, log)
+	if err := aws.Deploy(s, e, log); err != nil {
+		return err
+	}
+	if e.Provider.HostedZone != "" {
```

Comment:
in some parts of the code we use e.Root.HostedZone, in others this one. which is the correct one? do you think we should make it more obvious (maybe have two different objects) so the compilers catches it?

Reply:
It can be one or the other. User provided DNS is higher priority